lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
In my opinion, we should review merged code just as often as new code, but
that's besides the point.

The correct name of the encoding is "UTF-8" as defined in the standards:

https://www.ietf.org/rfc/rfc2279.txt
http://standards.iso.org/ittf/PubliclyAvailableStandards/index.html
(ISO/IEC 10646)
https://en.wikipedia.org/wiki/UTF-8#External_links

libiconv, for example, won't accept "utf8" as an encoding at all. If we had
extracted this constant earlier, this typo would not have occurred.

http://git.savannah.gnu.org/cgit/libiconv.git/tree/lib/encodings.def?id=48f31c744eb14da699c6ba3fd03f5f232c476bbb#n60

On Mon, May 9, 2016 at 5:53 PM, Balazs Scheidler <notifications@github.com>
wrote:

> In lib/logproto/logproto-server.c
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62523578>:
>
> >    self->encoding = g_strdup(encoding);
> > +
> > +  /* validate encoding */
> > +  convert = g_iconv_open("utf8", encoding);
>
> Hmm, this is already merged. But I don't agree that it matters how we call
> utf8. There are aliases for that for a reason. It doesn't make stuff more
> readable, reliable or anything. On the other hand it takes away energy to
> look after/validate these kind of mistakes we would make while coding.
> … <#m_-3246939082143047282_>
> On May 9, 2016 5:36 PM, "Tamas Nagy" ***@***.***> wrote: In
> lib/logproto/logproto-server.c <#797 (comment)
> <https://github.com/balabit/syslog-ng/pull/797#discussion_r62520468>>: >
> self->encoding = g_strdup(encoding); > + > + /* validate encoding */ > +
> convert = g_iconv_open("utf8", encoding); It would look nicer if we used
> UTF-8 everywhere, and we could introduce a constant for that as well. — You
> are receiving this because you were mentioned. Reply to this email directly
> or view it on GitHub <
> https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62520468>
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/797/files/35c68349ce999d0318e59abcb3cf4cccef630c81..3fa60288548b54bcfad7b42deb145ef2eafa029c#r62523578>
>